src/coreclr/System.Private.CoreLib/src/System/Reflection/Emit/RuntimeAssemblyBuilder.cs

```diff
@@ -299,5 +299,7 @@ protected override void SetCustomAttributeCore(ConstructorInfo con, ReadOnlySpan
                     binaryAttribute);
             }
         }
+
+        protected override void SaveCore(Stream stream) => throw new NotSupportedException();
```

Comment:
How would someone hit this? If it's just via the new public Save method on AssemblyBuilder, can we put a more descriptive error message into the exception that tells the user what they should be doing instead to create the builder?

Reply:
> How would someone hit this? 

If one creates and `AssemblyBuilder` using `DefineDynamicAssembly` overloads instead of `DefinePersistedAssembly` and call `AssemblyBuilder.Save(...)`.

> If it's just via the new public Save method on AssemblyBuilder, can we put a more descriptive error message into the exception that tells the user what they should be doing instead to create the builder?

Sure, good point